lib/code_owners.rb

```diff
@@ -35,11 +38,25 @@ def search_codeowners_file
     end
 
     # read the github file and spit out a slightly formatted list of patterns and their owners
+    # Empty/invalid/commented lines are still included in order to preserve line numbering
     def pattern_owners
       codeowner_path = search_codeowners_file
-      File.read(codeowner_path).split("\n").map do |line|
-        line.gsub(/#.*/, '').gsub(/^$/, " @").split(/\s+@/, 2)
-      end
+      i = 0
+      patterns = []
+      File.read(codeowner_path).split("\n").each { |line|
```

Comment:
can replace `each` with `each_with_index` and drop the i setting and updater

Reply:
Ah thanks! I was searching for something similar to Python's `enumerate` but was failing to grasp at the correct terms / find the right documentation.